test/exchange-files.js

```diff
@@ -125,31 +122,27 @@ describe('exchange files', () => {
         daemon2 = nodes[1]
       })
 
-      before('connect', function (done) {
+      before('connect', async function () {
         this.timeout(timeout)
 
-        series([
-          (cb) => parallel([
-            (cb) => daemon1.api.id(cb),
-            (cb) => daemon2.api.id(cb)
-          ], (err, ids) => {
-            expect(err).to.not.exist()
-            id1 = ids[0]
-            id2 = ids[1]
-            cb()
-          }),
-          (cb) => daemon1.api.swarm.connect(id2.addresses[0], cb),
-          (cb) => daemon2.api.swarm.connect(id1.addresses[0], cb),
-          (cb) => parallel([
-            (cb) => daemon1.api.swarm.peers(cb),
-            (cb) => daemon2.api.swarm.peers(cb)
-          ], (err, peers) => {
-            expect(err).to.not.exist()
-            expect(peers[0].map((p) => p.peer.toB58String())).to.include(id2.id)
-            expect(peers[1].map((p) => p.peer.toB58String())).to.include(id1.id)
-            cb()
-          })
-        ], done)
+        const ids = await Promise.all([
+          daemon1.api.id(),
+          daemon2.api.id()
+        ])
+
+        id1 = ids[0]
+        id2 = ids[1]
+
+        await daemon1.api.swarm.connect(id2.addresses[0])
+        await daemon2.api.swarm.connect(id1.addresses[0])
+
+        const peers = await Promise.all([
```

Comment:
This can also be destructured:

```javascript
const [ peers1, peers2 ] = await Promise.all([
  daemon1.api.swarm.peers(),
  daemon2.api.swarm.peers()
])

expect(peers1.map((p) => p.peer.toB58String())).to.include(id2.id)
expect(peers2.map((p) => p.peer.toB58String())).to.include(id1.id)
```